azure-iot-device/azure/iot/device/iothub/pipeline/iothub_pipeline.py

```diff
@@ -21,15 +21,26 @@
 )
 from azure.iot.device.iothub.auth.x509_authentication_provider import X509AuthenticationProvider
 
+from azure.iot.device.iothub.config import IoTHubPipelineConfig
+
 logger = logging.getLogger(__name__)
 
 
 class IoTHubPipeline(object):
-    def __init__(self, auth_provider):
+    def __init__(self, auth_provider, **kwargs):
         """
         Constructor for instantiating a pipeline adapter object
         :param auth_provider: The authentication provider
+        :param protocol: The protocol for connecting to IoT Hub
         """
+        try:
+            # Question: should this be kept in self? Is that necessary?
+            self._config = IoTHubPipelineConfig(**kwargs)
```

Comment:
>self._config = IoTHubPipelineConfig(**kwargs) [](start = 12, length = 45)

Per your question, no it shouldn't. Not necessary to keep a reference to it hanging around here. At least for now.

Reply:
kk